devices/libsdio/README.md

```diff
@@ -0,0 +1,277 @@
+| [in] `uint16_t blocksz` | block size for bulk transfers |
+
+> **NOTE:** `uint32_t freq` can only be specified as either 25000000 or 50000000, since only 25 and 50 MHz clock speeds are supported.
+
+Returns:
+
+| code | description |
+|------|-------------|
+|  0   | success     |
+| -1   | failure     |
+
+> **NOTE:** Every failure is accompanied by explicit fault description printed into standard output.
+
+---
+
+```c
+int sdio_transferDirect(sdio_dir_t dir, uint32_t address, uint8_t area, uint8_t *data);
+```
+
+This function initiates a direct, single 8-bit register read/write opetation. When `dir` is specified as `sdio_write` the byte inside the `data` buffer is written to the card, otherwise if `dir` equals `sdio_read` is specified, the buffer is set to value read from the card.
+
+Parameters:
+
+| parameter                | description                       |
+|--------------------------|-----------------------------------|
+| [in] `sdio_dir_t dir`    | transfer direction                |
+| [in] `uint32_t address`  | card register address to access   |
+| [in] `uint8_t area`      | card I/O area index to access     |
+| [in/out] `uint8_t *data` | bi-directional single byte buffer |
+
+Returns:
+
+| code | description |
+|------|-------------|
+|  0   | success     |
+| -1   | failure     |
+
+> **NOTE:** Every failure is accompanied by explicit fault description printed into standard output.
+
+---
+
+```c
+int sdio_transferBulk(sdio_dir_t dir, int blockMode, uint32_t address, uint8_t area, uint8_t *data, size_t len);
+```
+
+This function initiates an indirect, multi-byte transfer of up to 2048 bytes at once. As is the case with `sdio_transferDirect`, this call can service bi-directional transfers.
+
+Parameters:
+
+| parameter                | description                             |
+|--------------------------|-----------------------------------------|
+| [in] `sdio_dir_t dir`    | transfer direction                      |
+| [in] `int blockMode`     | wheter to use blocks for transfer (1/0) |
+| [in] `uint32_t address`  | card base address to access             |
+| [in] `uint8_t area`      | card I/O area index to access           |
+| [in/out] `uint8_t *data` | bi-directional multi byte buffer        |
+| [in] `size_t len`        | total transfer size in bytes            |
+
+> **NOTE:** `address` parameter specifies only the base address of the transfer. If `blockMode` is set to 1, then the address is automatically incremented by the card with every following block.
+
+Returns:
+
+| code        | description                                                 |
+|-------------|-------------------------------------------------------------|
+|  0          | success                                                     |
+| -1          | failure                                                     |
+| EINVAL (22) | `blocksz` is unset or `len` is not a multiple of block size |
```

Comment:
Uhh, we should not mix plain -1 and errnos. Also, don't we return -EINVAL (-22) instead?

Reply:
The issue was addressed already.